### PR TITLE
[marvell-arm64]: Enable SDK module for bookworm

### PR DESCRIPTION
**Pull request by Keshavg-marvell**
This patch enables compiling of Marvell platform
module and fixes sonic-platform-nokia compilation
issues for bookworm.

#### Why I did it

For enabling bookworm build on Marvell-arm64 and fixing https://github.com/sonic-net/sonic-buildimage/issues/16734

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Port Marvell SDK driver for bookworm  6.1 kernel.
Port Nokia CPLD driver for bookworm 6.1 kernel 
#### How to verify it
Built image from bookworm branch and installed on Marvell platform to 
verify "show interface status".

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by Keshavg-marvell** [comment]
@Pavan-Nokia please review

**Review by Pavan-Nokia — APPROVED** [review]

File: build_debian.sh [hunk 1]
```diff
@@ -747,8 +747,7 @@ if [[ $TARGET_BOOTLOADER == uboot ]]; then
         sudo LANG=C chroot $FILESYSTEM_ROOT mv /boot/u${INITRD_FILE} /boot/$INITRD_FILE
     elif [[ $CONFIGURED_ARCH == arm64 ]]; then
         sudo cp -v $PLATFORM_DIR/${sonic_asic_platform}-${CONFIGURED_ARCH}/sonic_fit.its $FILESYSTEM_ROOT/boot/
-		# TODO: fix for bookworm, probably needs arm64 patches in sonic-linux-kernel to be updated
-        #sudo LANG=C chroot $FILESYSTEM_ROOT mkimage -f /boot/sonic_fit.its /boot/sonic_${CONFIGURED_ARCH}.fit
+        sudo LANG=C chroot $FILESYSTEM_ROOT mkimage -f /boot/sonic_fit.its /boot/sonic_${CONFIGURED_ARCH}.fit
```
Comment (saiarcot895) [inline build_debian.sh:750]: This command is failing with `FATAL ERROR: Couldn't open "/usr/lib/linux-image-5.10.0-23-2-arm64/marvell/ac5-98dx35xx-rd.dtb": No such file or directory`. I think some PRs in sonic-linux-kernel need to be updated/enabled first?

Comment (Keshavg-marvell) [reply build_debian.sh:750]: looks like below line is missed in your last commit of changing the kernel version.  

https://github.com/sonic-net/sonic-buildimage/blob/bookworm/platform/marvell-arm64/sonic_fit.its#L23

Comment (saiarcot895) [reply build_debian.sh:750]: Ah, would you be able to make that change here?

Also, I see that you have also opened sonic-net/sonic-linux-kernel#352. That would probably fix the dtb file error.

Comment (Keshavg-marvell) [reply build_debian.sh:750]: @saiarcot895 I was on leave from last 6days hence the delay in response.

I have taken care this  "https://github.com/sonic-net/sonic-buildimage/blob/bookworm/platform/marvell-arm64/sonic_fit.its#L23" change and updated the PR. 

yes, https://github.com/sonic-net/sonic-linux-kernel/pull/352 would take care of dtb file errors.

Comment (saiarcot895) [reply build_debian.sh:750]: Can you include the sonic-linux-kernel submodule update in this PR?

Comment (Keshavg-marvell) [reply build_debian.sh:750]: Thanks Saikrishna for looking into this PR. I have updated sonic-linux-kernel submodule to this PR.

**Comment by Keshavg-marvell** [comment]
I believe it is good to merge now. Let us know if anything is needed from our side to merge these changes.

**Review by saiarcot895 — APPROVED** [review]

**Comment by saiarcot895** [comment]
@Keshavg-marvell Can armhf be enabled as well?

**Comment by Keshavg-marvell** [comment]
@saiarcot895 armhf we are planning to address with separate PR.

**Comment by saiarcot895** [comment]
Got it, is there an ETA for that PR?

**Comment by Keshavg-marvell** [comment]
> Got it, is there an ETA for that PR?

<keshav-marvell> ETA which we are internally tracking is 7th Nov.

**Comment by liushilongbuaa** [comment]
@saiarcot895 , please try again.

**Comment by saiarcot895** [comment]
New pipeline manually run [here](https://dev.azure.com/mssonic/build/_build/results?buildId=401321&view=results), where all checks pass.

**Comment by liushilongbuaa** [comment]
@saiarcot895 , Comment only works for Keshavg-marvell, the PR owner.
It is by design.

**Pull request merged**